### PR TITLE
Update test.py script for loading trained model and inferring with it.

### PR DESCRIPTION
**Pull request by eleven-yun**

### CODE REVIEW

File: scripts/test.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import os
+import sys
+import torch
+import torch.nn as nn
+
+cwd = os.getcwd()  # current working directory
+cfp = os.path.dirname(os.path.abspath(__file__))  # current file path
+os.chdir(cfp)
+sys.path.append("..")
```
Comment (LingerDavid) [inline scripts/test.py]: `sys.path.append("..")`->``sys.path.append(os.path.abspath(".."))``

File: scripts/test.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+from nn_modules.transformer.top_former import TopFormer
+os.chdir(cwd)
+
+top_former = TopFormer(num_layers=6, d_model=512, nhead=8, num_classes=3)
```
Comment (LingerDavid) [inline scripts/test.py:16]: Why not add the constants to configuration.

File: scripts/test.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+model_file_path = 'top_eleven.pt'
+
+# TODO: remove this line
+torch.save(top_former.state_dict(), model_file_path)
+
+top_former.load_state_dict(torch.load(model_file_path))
+top_former.eval()
+
+# TODO: replace with real data
+x = torch.rand(50, 1, 512)
```
Comment (LingerDavid) [inline scripts/test.py:31]: Add the constants to the configurations.

File: scripts/test.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+# TODO: remove this part
+for param in top_former.parameters():
+    if param.dim() > 1:
+        nn.init.xavier_uniform_(param)
+
+model_file_path = 'top_eleven.pt'
+
+# TODO: remove this line
+torch.save(top_former.state_dict(), model_file_path)
```
Comment (LingerDavid) [inline scripts/test.py:25]: More file check to avoid interrupted by crash.

File: scripts/test.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+top_former.load_state_dict(torch.load(model_file_path))
+top_former.eval()
+
+# TODO: replace with real data
+x = torch.rand(50, 1, 512)
+u = torch.rand(10, 1, 512)
+
+p = top_former(x, u)
```
Comment (LingerDavid) [inline scripts/test.py]: The shape of p is (10,3)?

File: scripts/test.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+# TODO: replace with better interface
+print(p)
```
Comment (LingerDavid) [inline scripts/test.py]: More elegant visualization for output

File: scripts/test.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    if param.dim() > 1:
+        nn.init.xavier_uniform_(param)
+
+model_file_path = 'top_eleven.pt'
+
+# TODO: remove this line
+torch.save(top_former.state_dict(), model_file_path)
+
+top_former.load_state_dict(torch.load(model_file_path))
+top_former.eval()
```
Comment (LingerDavid) [inline scripts/test.py:28]: Without autograd

File: scripts/test.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+cfp = os.path.dirname(os.path.abspath(__file__))  # current file path
+os.chdir(cfp)
+sys.path.append("..")
+from nn_modules.transformer.top_former import TopFormer
+os.chdir(cwd)
+
+top_former = TopFormer(num_layers=6, d_model=512, nhead=8, num_classes=3)
+
+# TODO: remove this part
+for param in top_former.parameters():
```
Comment (LingerDavid) [inline scripts/test.py]: Encapsulate the logic into a function

File: scripts/test.py (hunk 1 above)
Comment (eleven-yun) [reply scripts/test.py]: Done

File: scripts/test.py (hunk 2 above)
Comment (eleven-yun) [reply scripts/test.py:16]: To be addressed in a follow-up patch.

File: scripts/test.py (hunk 3 above)
Comment (eleven-yun) [reply scripts/test.py:31]: To be addressed in a follow-up patch.

File: scripts/test.py (hunk 8 above)
Comment (eleven-yun) [reply scripts/test.py]: Code refactoring done.

File: scripts/test.py (hunk 7 above)
Comment (eleven-yun) [reply scripts/test.py:28]: Good point. Done.

Inference gets faster.

File: scripts/test.py (hunk 5 above)
Comment (eleven-yun) [reply scripts/test.py]: The shape of `p` is `(10, 1, 3)`, where `10` is decided by the decoder input `u`, according to the current design of our model struct.

We probably need to revise the design of our model struct later.

File: scripts/test.py (hunk 6 above)
Comment (eleven-yun) [reply scripts/test.py]: Done. Used `tabulate` lib for nicer print-out.

File: scripts/test.py (hunk 4 above)
Comment (eleven-yun) [reply scripts/test.py:25]: Done.

A helper function `check_model_file_path()` is added to `utils`.

**Review by eleven-yun — COMMENTED** [review]
All review comments addressed.

**Review by LingerDavid — APPROVED** [review]

**Pull request merged**